sql/updates/latest-dev.sql

```diff
@@ -21,3 +21,39 @@ ALTER TABLE _timescaledb_catalog.chunk
   ALTER COLUMN  osm_chunk SET DEFAULT FALSE;
 
 CREATE INDEX chunk_osm_chunk_idx ON _timescaledb_catalog.chunk (osm_chunk, hypertable_id);
+
+-- cagg migration catalog relations
+CREATE TABLE _timescaledb_catalog.continuous_agg_migrate_plan (
+  mat_hypertable_id integer NOT NULL,
+  start_ts TIMESTAMPTZ NOT NULL DEFAULT pg_catalog.now(),
+  end_ts TIMESTAMPTZ,
+  -- table constraints
+  CONSTRAINT continuous_agg_migrate_plan_pkey PRIMARY KEY (mat_hypertable_id),
+  CONSTRAINT continuous_agg_migrate_plan_mat_hypertable_id_fkey FOREIGN KEY (mat_hypertable_id) REFERENCES _timescaledb_catalog.continuous_agg (mat_hypertable_id)
+);
+
+SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_agg_migrate_plan', '');
+
+CREATE TABLE _timescaledb_catalog.continuous_agg_migrate_plan_step (
+  mat_hypertable_id integer NOT NULL,
+  step_id serial NOT NULL,
+  status TEXT NOT NULL DEFAULT 'NOT STARTED', -- NOT STARTED, STARTED, FINISHED, CANCELED
+  start_ts TIMESTAMPTZ,
+  end_ts TIMESTAMPTZ,
+  type TEXT NOT NULL,
+  config JSONB,
+  -- table constraints
+  CONSTRAINT continuous_agg_migrate_plan_step_pkey PRIMARY KEY (mat_hypertable_id, step_id),
+  CONSTRAINT continuous_agg_migrate_plan_step_mat_hypertable_id_fkey FOREIGN KEY (mat_hypertable_id) REFERENCES _timescaledb_catalog.continuous_agg_migrate_plan (mat_hypertable_id) ON DELETE CASCADE,
+  CONSTRAINT continuous_agg_migrate_plan_step_check CHECK (start_ts <= end_ts),
+  CONSTRAINT continuous_agg_migrate_plan_step_check2 CHECK (type IN ('CREATE NEW CAGG', 'DISABLE POLICIES', 'COPY POLICIES', 'ENABLE POLICIES', 'SAVE WATERMARK', 'REFRESH NEW CAGG', 'COPY DATA'))
+);
+
+SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_agg_migrate_plan_step', '');
+
+SELECT pg_catalog.pg_extension_config_dump(pg_get_serial_sequence('_timescaledb_catalog.continuous_agg_migrate_plan_step', 'step_id'), '');
+
+-- in tables.sql the same is done with GRANT SELECT ON ALL TABLES IN SCHEMA
+GRANT SELECT ON _timescaledb_catalog.continuous_agg_migrate_plan TO PUBLIC;
+GRANT SELECT ON _timescaledb_catalog.continuous_agg_migrate_plan_step TO PUBLIC;
+GRANT SELECT ON _timescaledb_catalog.continuous_agg_migrate_plan_step_step_id_seq TO PUBLIC;
```

Comment:
```suggestion
GRANT SELECT ON pg_get_serial_sequence('_timescaledb_catalog.continuous_agg_migrate_plan_step', 'step_id') TO PUBLIC;
```
I don't know if we can do this, or if it is even necessary to do the GRANT on the serial. In any case, if we have to use the name `continuous_agg_migrate_plan_step_step_id_seq` anyway, why do we use `pg_get_serial_sequence('_timescaledb_catalog.continuous_agg_migrate_plan_step', 'step_id')` in the first place?

Reply:
The syntax you're proposing is not allowed in Postgres, you should specify the object name in GRANT statement: https://www.postgresql.org/docs/current/sql-grant.html

Have a look if we try to run what you're proposing:

```sql
fabrizio=# GRANT SELECT ON pg_get_serial_sequence('_timescaledb_catalog.continuous_agg_migrate_plan_step', 'step_id') TO PUBLIC;
ERROR:  syntax error at or near "("
LINE 1: GRANT SELECT ON pg_get_serial_sequence('_timescaledb_catalog...
```

Reply:
> In any case, if we have to use the name continuous_agg_migrate_plan_step_step_id_seq anyway

That's why then we should be consistent, if we use `continuous_agg_migrate_plan_step_step_id_seq` we have to remove `pg_get_serial_sequence('_timescaledb_catalog.continuous_agg_migrate_plan_step', 'step_id')`

Reply:
Make sense... fixed!